csharp/client/Dh_NetClient/container/RowSequence.cs

```diff
@@ -0,0 +1,174 @@
+﻿//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+using System;
+
+namespace Deephaven.Dh_NetClient;
+
+public abstract class RowSequence {
+  public static RowSequence CreateEmpty() => SequentialRowSequence.EmptyInstance;
+
+  public static RowSequence CreateSequential(Interval interval) {
+    return new SequentialRowSequence(interval);
+  }
+
+  public bool IsEmpty => Count == 0;
+  public abstract UInt64 Count { get; }
+
+  public abstract IEnumerable<Interval> Intervals { get; }
+
+  public IEnumerable<UInt64> Elements {
+    get {
+      foreach (var interval in Intervals) {
+        for (UInt64 i = interval.Begin; i != interval.End; ++i) {
+          yield return i;
+        }
+      }
+    }
+  }
+
+  public abstract RowSequence Take(UInt64 size);
+  public abstract RowSequence Drop(UInt64 size);
+}

```

Comment:
The implementations of `RowSequence` are all immutable, is that the planned design?

Reply:
Yes. Every time I make a data structure immutable, an angel gets its wings.